### PR TITLE
Added diagram for alter type

### PR DESCRIPTION
**Pull request by Amruta-Ranade**
Reference issue: https://github.com/cockroachdb/docs/issues/2992

### CODE REVIEW

File: pkg/cmd/docgen/diagrams.go [hunk 1]
```diff
@@ -377,6 +377,12 @@ var specs = []stmtSpec{
 		unlink:  []string{"table_name"},
 		nosplit: true,
 	},
+	{
```
Comment (mjibson) [inline pkg/cmd/docgen/diagrams.go]: Here's another option that avoids hard coding everything:

```
	{
		name:   "alter_type",
		stmt:   "alter_onetable_stmt",
		inline: []string{"alter_table_cmds", "alter_table_cmd", "opt_column", "opt_set_data"},
		match:  []*regexp.Regexp{regexp.MustCompile(`'ALTER' ('COLUMN')? column_name ('SET' 'DATA')? 'TYPE'`)},
		regreplace: map[string]string{
			regList:                  "",
			"opt_collate":            "",
			"opt_alter_column_using": "",
		},
		replace: map[string]string{
			"relation_expr": "table_name",
		},
		unlink: []string{"table_name", "column_name"},
	},
```

**Review by madelynnblue — APPROVED** [review]
LGTM, but you need to squash your commits before merging. I can show you how to do that if you like.

**Comment by Amruta-Ranade** [comment]
@mjibson Yes, I need help with the commit squashing..tried doing it myself and I think I made it worse.

**Comment by mjibson** [comment]
Ok I just fixed this by hand because it was weird. I think it's ready to be merged now.

**Pull request merged**